src/stores/queryStore/createQueryControllerSlice.ts

```diff
@@ -0,0 +1,71 @@
+import {StateCreator} from "zustand";
+
+import {LOG_LEVEL} from "../../typings/logs";
+import {DO_NOT_TIMEOUT_VALUE} from "../../typings/notifications";
+import useContextStore from "../contextStore";
+import useLogFileManagerStore from "../logFileManagerProxyStore";
+import {QUERY_RESULTS_DEFAULT} from "./createQueryResultsSlice";
+import {QUERY_CONFIG_DEFAULT} from "./queryConfigSlice";
+import {
+    QueryControllerSlice,
+    QueryControllerValues,
+    QueryState,
+} from "./types";
+
+
+const QUERY_CONTROLLER_DEFAULT: QueryControllerValues = {
+    queryProgress: 0,
+};
+
+/**
+ * Creates a slice for handling queries & clearing queries.
+ *
+ * @param set
+ * @param get
+ * @return
+ */
+const createQueryControllerSlice: StateCreator<
+    QueryState, [], [], QueryControllerSlice
+> = (set, get) => ({
+    ...QUERY_CONTROLLER_DEFAULT,
+    clearQuery: () => {
+        set({
+            ...QUERY_CONFIG_DEFAULT,
+            ...QUERY_CONTROLLER_DEFAULT,
+            ...QUERY_RESULTS_DEFAULT,
+        });
+    },
+    setQueryProgress: (newProgress) => {
+        set({queryProgress: newProgress});
+    },
+    startQuery: () => {
+        const {
+            clearQueryResults,
+            queryString,
+            queryIsCaseSensitive,
+            queryIsRegex,
+        } = get();
+        const {logFileManagerProxy} = useLogFileManagerStore.getState();
+        const {postPopUp} = useContextStore.getState();
+
+        if (QUERY_CONFIG_DEFAULT.queryString === queryString) {
+            return;
```

Comment:
I believe we need to clearResults() before early returning. Otherwise, any results from previous queries will still be rendered in the panel even if the query string is empty. e.g.,
![image](https://github.com/user-attachments/assets/287ed953-5c0c-4dc1-a9cd-fe5ad7c285a5)

@coderabbitai please create an issue to track this